pkg/kube/pod.go

```diff
@@ -299,7 +299,7 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 		}
 		switch p.Status.Phase {
 		case v1.PodFailed:
-			return false, errors.Errorf("Pod %s failed. Pod details (%s)", name, log.SafeDumpPodObject(p))
+			return false, errors.Errorf("Pod %s failed. Pod Status details (%s)", name, log.SafeDumpPodStatusObject(p))
```

Comment:
We can just use `p.Status.String()` directly. Changes to helper and the tests will not be necessary

Reply:
Agreed - then we can also remove `log.SafeDumpPodObject()`.